src/Http/ExpoController.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace NotificationChannels\ExpoPushNotifications\Http;
+
+use Illuminate\Http\Request;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Routing\Controller;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Validator;
+use NotificationChannels\ExpoPushNotifications\ExpoChannel;
+
+class ExpoController extends Controller
+{
+    /**
+     * @var ExpoChannel
+     */
+    private $expoChannel;
+
+    /**
+     * ExpoController constructor.
+     *
+     * @param ExpoChannel $expoChannel
+     */
+    public function __construct(ExpoChannel $expoChannel)
+    {
+        $this->expoChannel = $expoChannel;
+    }
+
+    /**
+     * Handles subscription endpoint for an expo token.
+     *
+     * @param Request $request
+     *
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function subscribe(Request $request)
+    {
+        $validator = Validator::make($request->all(), [
+            'expo_token'    =>  'required|string',
+        ]);
+
+        if ($validator->fails()) {
+            return JsonResponse::create([
+                'status' => 'failed',
+                'error' => [
+                    'message' => 'Expo Token is required',
+                ],
+            ], 422);
+        }
+
+        $token = $request->get('expo_token');
+
+        $interest = $this->expoChannel->interestName(Auth::user());
+
+        try {
+            $this->expoChannel->expo->subscribe($interest, $token);
+        } catch (\Exception $e) {
+            return JsonResponse::create([
+                'status'    => 'failed',
+                'error'     =>  [
+                    'message' => $e->getMessage(),
```

Comment:
Since we are catching `\Exception` is it possible this could/will have sensitive info in it?